DesktopUI2/DesktopUI2/ViewModels/CollaboratorsViewModel.cs

```diff
@@ -63,44 +63,56 @@ internal void ReloadUsers()
 
   private async void Search()
   {
-    Focus();
-    if (SearchQuery.Length < 3)
+    try
     {
-      return;
-    }
+      Focus();
+      if (SearchQuery.Length < 3)
+      {
+        return;
+      }
 
-    if (!await Http.UserHasInternet().ConfigureAwait(true))
-    {
-      Dispatcher.UIThread.Post(
-        () =>
-          MainUserControl.NotificationManager.Show(
-            new PopUpNotificationViewModel
-            {
-              Title = "⚠️ Oh no!",
-              Message = "Could not reach the internet, are you connected?",
-              Type = NotificationType.Error
-            }
-          ),
-        DispatcherPriority.Background
-      );
+      if (!await Http.UserHasInternet().ConfigureAwait(true))
+      {
+        Dispatcher.UIThread.Post(
+          () =>
+            MainUserControl.NotificationManager.Show(
+              new PopUpNotificationViewModel
+              {
+                Title = "⚠️ Oh no!",
+                Message = "Could not reach the internet, are you connected?",
+                Type = NotificationType.Error
+              }
+            ),
+          DispatcherPriority.Background
+        );
 
-      return;
-    }
+        return;
+      }
 
-    if (SearchQuery.Contains("@"))
-    {
-      if (Utils.IsValidEmail(SearchQuery))
+      if (SearchQuery.Contains("@"))
       {
-        var emailAcc = new AccountViewModel { Name = SearchQuery };
-        Users = new List<AccountViewModel> { emailAcc };
+        if (Utils.IsValidEmail(SearchQuery))
+        {
+          var emailAcc = new AccountViewModel { Name = SearchQuery };
+          Users = new List<AccountViewModel> { emailAcc };
 
-        ShowProgress = false;
-        DropDownOpen = true;
+          ShowProgress = false;
+          DropDownOpen = true;
+        }
+      }
+      else
+      {
+        userSearchDebouncer();
       }
     }
-    else
+    catch (Exception ex) when (!ex.IsFatal())
```

Comment:
If we throw Fatal error up, is that going to cause the host app issues or will DUI handle it gracefully?

Reply:
I will remove this